hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -212,6 +234,9 @@ public KafkaOffsetGen(TypedProperties props) {
       Set<TopicPartition> topicPartitions = partitionInfoList.stream()
               .map(x -> new TopicPartition(x.topic(), x.partition())).collect(Collectors.toSet());
 
+      if (Config.KAFKA_CHECKPOINT_TYPE_TIMESTAMP.equals(kafkaCheckpointType) && isValidCheckpointType(lastCheckpointStr)) {
+        lastCheckpointStr = getOffsetsByTimestamp(consumer, partitionInfoList, topicPartitions, topicName, Long.parseLong(lastCheckpointStr.get()));
+      }
```

Comment:
I don't think there needs to be an "else if" here.
If you are using timestamp kafkaCheckpointType, lastCheckpointStr will be passed a timestamp of "timestamp type", which we will handle using the getOffsetByTimestamp method.
If it is not a timestamp type, then we can interpret it as a regular string type checkpoint rule and do not process it

Reply:
not sure I understand. this is what I am thinking
```
if (timestamp based checkpoint)
    lastCheckpoint = getOffsetByTimestamp()
else if regular checkpoint type
   lastCheckpoint = fetValidOffsets()
else 
   reset based on auto.offset.reset. 
```

Am I misunderstanding anything here?

Reply:
It can be understood that when checkTimestamptype is not used, the format of lastCheckpointStr is topic_name,partition_num:offset,partition_num:offset

When getOffsetsByTimestamp method is used, what we do is to convert lastCheckpointStr=timestamp to topic_name,partition_num:offset,partition_num:offset

Reply:
ok, I get it now. makes sense.